src/main/java/de/cas/mse/exercise/diamond/Diamond.java

```diff
@@ -3,29 +3,38 @@
 import java.util.Arrays;
 
 public class Diamond {
+    public static void main(String[] args) {
+        System.out.println(new Diamond().generateDiamond(5));
+    }
 
-	public String print(int n) {
-		if (n <= 0 || n % 2 == 0) {
+	public String generateDiamond(int height) {
+		if (!isValidDiamondShape(height)) {
 			return null;
+        }
+        String middleLine = new String(makeLine(height, height));
+        StringBuilder diamond = new StringBuilder(middleLine);
+        // iteratively expand diamond towards top and bottom
+		for (int asterisks = height - 2; asterisks > 0; asterisks -= 2) {
+			char[] line = makeLine(height, asterisks);
+			diamond.insert(0, line);
+			diamond.append(line);
 		}
-		StringBuilder builder = new StringBuilder(new String(make(n, n)));
-		for (int i = n - 2; i > 0; i -= 2) {
-			char[] chars = make(n, i);
-			builder.insert(0, chars);
-			builder.append(chars);
-		}
-		return builder.toString();
-	}
+		return diamond.toString();
+    }
+    
+    private boolean isValidDiamondShape(int height) {
+        return height <= 0 || height % 2 == 0;
+    }
 
-	private char[] make(int i, int j) {
-		int amount = ((i - j) / 2);
-		char[] chars = new char[amount + j + 1];
-		if (amount > 0) {
-			Arrays.fill(chars, 0, amount, ' ');
+	private char[] makeLine(int diamondWidth, int numberOfAsterisks) {
```

Comment:
Either change this into diamondHeight (which is how the method is called on line 18), or maxWidth to indicate that it's a fixed value.